kernel: kconfig: changing recommendations for SYS_CLOCK_TICKS_PER_SEC

yerabolu opened the pull request:
Documentation for kconfig SYS_CLOCK_TICKS_PER_SEC has some outdated
recommendations. Changing them to align with other documentation
under kernel timing.

Fixes: #25482

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

nashif commented on kernel/Kconfig:
```diff
@@ -502,30 +502,19 @@ config ARCH_HAS_CUSTOM_BUSY_WAIT
 
 config SYS_CLOCK_TICKS_PER_SEC
 	int "System tick frequency (in ticks/second)"
-	default 100 if QEMU_TARGET || SOC_POSIX
```
why are you removing those?

yerabolu replied:
sorry, that's a mistake. will update it now

nashif commented on include/kernel.h:
```diff
@@ -4607,6 +4607,8 @@ extern int k_mem_slab_init(struct k_mem_slab *slab, void *buffer,
  *
  * This routine allocates a memory block from a memory slab.
  *
+ * @note Can be called by ISRs, but @a timeout must be set to K_NO_WAIT.
```
how is this change related to the Kconfig changes in the title of this PR?

yerabolu replied:
They are not related, for some reason my last PR changes got submitted along with this one.  Pushed the changes.

yerabolu committed:
kernel: kconfig: changing recommendations for SYS_CLOCK_TICKS_PER_SEC

Documentation for kconfig SYS_CLOCK_TICKS_PER_SEC has some outdated
recommendations. Changing them to align with other documentation
under kernel timing.

Fixes: #25482

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

nashif commented on kernel/Kconfig:
```diff
@@ -508,24 +508,16 @@ config SYS_CLOCK_TICKS_PER_SEC
-	  configurations even require calls to compiler built-in functions, and
-	  can require a non-trivial extra amount of stack space (e.g. around 80
-	  bytes on x86).
+	  For asynchronous timekeeping, the kernel defines a "ticks" concept.  A
```
what used to be easy to read is now condensed in one paragraph. 
Also, "The choice of tick rate is configurable via :c:option:`CONFIG_SYS_CLOCK_TICKS_PER_SEC`" sound like redundant, because this is the same option we are documenting.

yerabolu replied:
Changed it to a 2 para and removed via :c:option:CONFIG_SYS_CLOCK_TICKS_PER_SEC

pabigot commented on kernel/Kconfig:
```diff
@@ -508,24 +508,16 @@ config SYS_CLOCK_TICKS_PER_SEC
+	  uptime and timeout bookeeping.  Interrupts are expected to be
+	  delivered on tick boundaries to the extent practical, and no
+	  fractional ticks are tracked.  The choice of tick rate is configurable
+	  via :c:option:`CONFIG_SYS_CLOCK_TICKS_PER_SEC`.  Defaults on most
```
AFAICT we don't ever use RST annotations like `:c:option` in Kconfig help text.

Agree that a paragraph break before "The choice of tick rate" would read better, and the sentence could be "The choice of tick rate is configured by this option.  Defaults on..."  

A sentence like "Also the number of cycles per tick should be chosen so that 1 millisecond is exactly represented by an integral number of ticks." would be worth adding somewhere in that second paragram, since if that is not true `k_uptime_get()` will accumulate error.

Otherwise this is a vast improvement, thanks.

yerabolu replied:
@pabigot : Addressed the changes, could you review this again?
